corehq/apps/users/models.py

```diff
@@ -1126,6 +1126,9 @@ def get(stale, raise_if_none):
     def clear_quickcache_for_user(self):
         from corehq.apps.hqwebapp.templatetags.hq_shared_tags import _get_domain_list
         self.get_by_username.clear(self.__class__, self.username)
+        self.get_by_user_id.clear(self.__class__, self.user_id)
+        for domain in self.get_domains():
+            self.get_by_user_id.clear(self.__class__, self.user_id, domain)
```

Comment:
the list of domains could have changed by the time we call this - i think we might have to keep track of what get_domains() returns on __init__ and clear the union between that set of domains and the current set of domains